mailpoet/lib/Mailer/Methods/MailPoet.php

```diff
@@ -82,14 +82,20 @@ public function send($newsletter, $subscriber, $extraParams = []): array {
   }
 
   public function processSendError($result, $subscriber, $newsletter) {
-    if (!empty($result['code']) && $result['code'] === API::RESPONSE_CODE_KEY_INVALID) {
-      $this->bridge->invalidateMssKey();
-    } elseif (
-      !empty($result['code'])
-      && $result['code'] === API::RESPONSE_CODE_CAN_NOT_SEND
-      && $result['error'] === API::ERROR_MESSAGE_INVALID_FROM
-    ) {
-      $this->authorizedEmailsController->checkAuthorizedEmailAddresses();
+    if (!empty($result['code'])) {
```

Comment:
There's something about the formatting of the conditionals that makes this a little hard for me to read. Do you have any guess about if we'll be adding to this in the future? If so (i.e. it might get even more complex) I might personally prefer something more like this (no guarantee that it's functionally the same!):

```php
public function processSendError($result, $subscriber, $newsletter) {
    if (empty($result['code'])) {
        return $this->errorMapper->getErrorForResult($result, $subscriber, $this->sender, $newsletter);
    }

    switch ($result['code']) {
        case API::RESPONSE_CODE_KEY_INVALID:
            $this->bridge->invalidateMssKey();
            break;

        case API::RESPONSE_CODE_CAN_NOT_SEND:
            if ($result['error'] === API::ERROR_MESSAGE_INVALID_FROM) {
                $this->authorizedEmailsController->checkAuthorizedEmailAddresses();
            }
            break;

        case API::RESPONSE_CODE_PAYLOAD_ERROR:
            if (!empty($result['error']) && $result['error'] === API::ERROR_MESSAGE_BULK_EMAIL_FORBIDDEN) {
                $this->authorizedEmailsController->checkAuthorizedEmailAddresses();
            }
            break;
    }

    return $this->errorMapper->getErrorForResult($result, $subscriber, $this->sender, $newsletter);
}
```

Reply:
Thanks for the suggestion. I will use the `switch`.

Reply:
Done in https://github.com/mailpoet/mailpoet/pull/5391/commits/0882a36c7bebb612b77833979953aa094038eebd